src/Medidata.ZipkinTracer.Core/SpanTracer.cs

```diff
@@ -81,11 +81,11 @@ public virtual Span SendClientSpan(string requestName, string traceId, string pa
             return newSpan;
         }
 
-        public virtual void ReceiveClientSpan(Span span, string clientServiceName)
+        public virtual void ReceiveClientSpan(Span span)
         {
             var annotation = new Annotation()
             {
-                Host = zipkinEndpoint.GetEndpoint(clientServiceName),
+                Host = span.Annotations[0].Host,
```

Comment:
Is it 100% sure that the span won't be null and it will also have a non-null Annotations collection with at least 1 element at any given time when this method is called? I would recommend to handle null span, null Annotations and empty Annotations collection cases (including but not necessarily error handling) before this statement (of course if you can ensure that there is no possibility for any of the cases above then there is no need for checks).